crates/libs/bindgen/src/rust/structs.rs

```diff
@@ -135,110 +135,48 @@ fn gen_windows_traits(writer: &Writer, def: metadata::TypeDef, name: &TokenStrea
+    if derive.is_empty() {
         quote! {}
     } else {
-        let name = ident.as_str();
-        let features = writer.cfg_features(cfg);
-
-        let fields = def.fields().filter_map(|f| {
-            if f.flags().contains(metadata::FieldAttributes::Literal) {
-                None
-            } else {
-                let name = f.name();
-                let ident = to_ident(name);
-                let ty = f.ty(Some(def));
-                if metadata::type_has_callback(&ty) {
-                    None
-                } else {
-                    Some(quote! { .field(#name, &self.#ident) })
-                }
-            }
-        });
-
         quote! {
-            #features
-            impl core::fmt::Debug for #ident {
-                fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
-                    f.debug_struct(#name) #(#fields)* .finish()
-                }
-            }
+            #[derive(#(#derive),*)]
         }
     }
 }
 
-fn gen_copy_clone(writer: &Writer, def: metadata::TypeDef, name: &TokenStream, cfg: &cfg::Cfg) -> TokenStream {
-    let features = writer.cfg_features(cfg);
-
-    if writer.sys || metadata::type_def_is_copyable(def) {
-        quote! {
-            #features
-            impl Copy for #name {}
-            #features
-            impl Clone for #name {
-                fn clone(&self) -> Self {
-                    *self
-                }
-            }
-        }
-    } else if def.class_layout().is_some() {
-        // Don't support copy/clone of packed structs: https://github.com/rust-lang/rust/issues/82523
+fn gen_clone(writer: &Writer, def: metadata::TypeDef, name: &TokenStream, cfg: &cfg::Cfg) -> TokenStream {
+    if writer.sys || metadata::type_def_is_copyable(def) || def.flags().contains(metadata::TypeAttributes::WindowsRuntime) || def.class_layout().is_some() {
         quote! {}
-    } else if !def.flags().contains(metadata::TypeAttributes::WindowsRuntime) {
-        quote! {
-            #features
-            impl Clone for #name {
-                fn clone(&self) -> Self {
-                    unsafe { core::mem::transmute_copy(self) }
-                }
-            }
-        }
     } else {
-        let fields = def.fields().map(|f| {
-            let name = to_ident(f.name());
-            if f.flags().contains(metadata::FieldAttributes::Literal) {
-                quote! {}
-            } else if metadata::field_is_blittable(f, def) {
-                quote! { #name: self.#name }
-            } else {
-                quote! { #name: self.#name.clone() }
-            }
-        });
+        let features = writer.cfg_features(cfg);
 
         quote! {
             #features
             impl Clone for #name {
                 fn clone(&self) -> Self {
-                    Self { #(#fields),* }
+                    unsafe { core::mem::transmute_copy(self) }
```

Comment:
I'd rather if you left the previous implementation, this feels incredibly dangerous....

Reply:
This is the previous implementation! 🥲 I've just moved it around some, and this is the last bit that's not derived. I have some more work to do to eliminate this but yes that's the goal. This PR didn't actually add any new manual implementations. This is just the last implementation that remains, for now.

Reply:
Ah, I see now, GitHub's diffing was confusing me :)